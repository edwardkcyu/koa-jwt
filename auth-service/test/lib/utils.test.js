const { hash } = require('../../src/lib/utils');

describe('utils.hash', () => {
  test('correctly create SHA256 hash', () => {
    expect(hash('abcd1234password')).toMatchSnapshot();
  });

  test('throw error for null input', () => {
    expect(() => {
      hash(null);
    }).toThrowErrorMatchingSnapshot();
  });

  test('throw error for undefined input', () => {
    expect(() => {
      hash(undefined);
    }).toThrowErrorMatchingSnapshot();
  });
});
