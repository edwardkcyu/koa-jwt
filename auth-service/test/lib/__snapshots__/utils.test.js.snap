// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.hash correctly create SHA256 hash 1`] = `"cba0021ca36477ed4be95626316ac66e31ef7d73e8f9e3888e36737e1a9ae597"`;

exports[`utils.hash throw error for null input 1`] = `"Error creating hash value"`;

exports[`utils.hash throw error for undefined input 1`] = `"Error creating hash value"`;
